### ISSUE TITLE
Ability to mock PushClient()

### ISSUE THREAD

**Issue by Bryksin** [jav/expo-server-sdk-java#4]
Hi

We use Spring Boot for our back-end, and based on the active profile (dev, qa, prod) we would need to either send real notification or fake it.

For faking it, we just run `okhttp3:mockwebserver` and pre-set mock response at the specific request path, however, to allow us to mock server and use is instead of real Expo we need to use our custom URL.

In your `PushClient` - `BASE_URL` is set to final and there are is no ability to overwrite it.

Would be very friendly if you would allow second `PushClient()` constructor which will take String as argument and set it to `BASE_URL`

To be super nice, you could also allow overwriting `BASE_API_URL`

**Comment by jav** [comment]
Hi!

I've released version 0.5.4, which, to the best of my understanding, should solve your issue.

I've removed `BASE_URL` and `BASE_URL_API` and replace with only `baseUrlApi`, with get/set methods.

This way, the url dosn't build on the previous assumption that it is a domain + a path, but rather an arbitrarily set string.

I added the constructor you requested _and_ `@Deprecate`:ed it immediately.
I intend to move the class patterns towards fewer constructors and with lower arity in favor of builder pattern.

An example can be found in `PushClient#apiBaseUrlIsOverridable`  (there's a similar one in `PUshClient#getOneReceipt` )
```
        PushClient client = new PushClient();
        ...
        String mockBaseApiUrl = "mockString";
        client.setBaseApiUrl(mockBaseApiUrl);
        ...
```


Does this address the issue you where having?
